### PR TITLE
feat(swagger): add documentsEnabled option to disable JSON/YAML

### PR DESCRIPTION
**Pull request by mag123c**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?
Currently, the Swagger module always serves both Swagger UI and JSON/YAML API definitions. There is no way to selectively disable one or the other.

Issue Number: #3157 

## What is the new behavior?
This PR introduces two new options to provide greater control over Swagger endpoints:

swaggerUiEnabled: If false, the Swagger UI will not be served.
documentsEnabled: If false, JSON and YAML API definitions will not be served.

Developers can now:
1. Disable the Swagger UI but retain JSON/YAML definitions.
2. Disable JSON/YAML definitions but retain the Swagger UI.
3. Completely disable all Swagger-related endpoints.

These changes enhance the flexibility and security of Swagger integration, enabling better alignment with deployment needs.

JSON/YAML is not provided as below when documentEnabled is set to FALSE.

image
image

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: lib/interfaces/swagger-custom-options.interface.ts [hunk 1]
```diff
@@ -10,12 +10,20 @@ export interface SwaggerCustomOptions {
...
    * Default: `true`.
    */
   swaggerUiEnabled?: boolean;
 
+  /**
+   * If `false`, both the Swagger UI and API definitions (JSON and YAML) will be disabled.
+   * Use this option when you want to completely hide all Swagger-related endpoints.
+   * Default: `true`.
+   */
+  documentsEnabled?: boolean;
```
Comment (kamilmysliwiec) [inline lib/interfaces/swagger-custom-options.interface.ts:25]:
```suggestion
  raw?: boolean | Array<'json' | 'yaml'>;
```
could we rename it to "raw"? raw: true means, `@nestjs/swagger` will serve raw definitions. We could also allow passing an array that determines what formats should be served, e.g. `raw: ['json']` (meaning, no yaml).

File: lib/interfaces/swagger-custom-options.interface.ts [hunk 2]
```diff
@@ -10,12 +10,22 @@ export interface SwaggerCustomOptions {
   useGlobalPrefix?: boolean;
 
   /**
-   * If `false`, only API definitions (JSON and YAML) will be served (on `/{path}-json` and `/{path}-yaml`).
-   * This is particularly useful if you are already hosting a Swagger UI somewhere else and just want to serve API definitions.
+   * If `false`, the Swagger UI will not be served. Only API definitions (JSON and YAML)
+   * will be accessible (on `/{path}-json` and `/{path}-yaml`).
+   * To fully disable both the Swagger UI and API definitions, use `documentsEnabled: false`.
```
Comment (kamilmysliwiec) [inline lib/interfaces/swagger-custom-options.interface.ts:15]:
```suggestion
   * To fully disable both the Swagger UI and API definitions, use `raw: false`.
```

**Comment by kamilmysliwiec** [comment]
LGTM

**Comment by mag123c** [comment]
@kamilmysliwiec  
I didn't revise the official document after this PR, and I just requested PR to revise the official document.
with the deprecated (replacing with Ui) of swaggerUiEnabled.

https://github.com/nestjs/swagger/commit/675a0272ca3d60db716ee97bf17012dddfe38637

**Pull request merged**